src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1339,20 +1365,13 @@ protected override void InvokeDefaultAction(string path)
                 catch (Win32Exception ex) when (ex.NativeErrorCode == 13)
                 {
                     // Error code 13 -- Permission denied.
-                    // The file is possibly not an executable, so we try invoking the default program that handles this file.
-                    const string quoteFormat = "\"{0}\"";
-                    invokeProcess.StartInfo.FileName = Platform.IsLinux ? "xdg-open" : /* OS X */ "open";
-                    if (NativeCommandParameterBinder.NeedQuotes(path))
-                    {
-                        path = string.Format(CultureInfo.InvariantCulture, quoteFormat, path);
-                    }
-                    invokeProcess.StartInfo.Arguments = path;
-                    invokeProcess.Start();
+                    StartProcessWithOpen(path);
                 }
 #elif CORECLR
-                catch (Win32Exception ex) when (ex.NativeErrorCode == 193)
+                catch (Win32Exception ex) when (ex.NativeErrorCode == 193 || ex.NativeErrorCode == 5)
```

Comment:
Actually, I think this will actually make the code a bit more complex as I have to check if it's a Directory & not Nano/IoT (headless) for Windows.  The exception is handling cases where you invoke a .gif or .doc so that also needs to check.